axum/src/extract/path/de.rs

```diff
@@ -94,7 +94,20 @@ impl<'de> Deserializer<'de> for PathDeserializer<'de> {
                 .got(self.url_params.len())
                 .expected(1));
         }
-        visitor.visit_borrowed_str(&self.url_params[0].1)
+        let key = &self.url_params[0].0;
+        let value = &self.url_params[0].1;
+        visitor
+            .visit_borrowed_str(value)
+            .map_err(|e: Self::Error| match e.kind {
+                ErrorKind::Message(message) => {
+                    PathDeserializationError::new(ErrorKind::DeserializeError {
+                        key: key.to_string(),
+                        value: value.as_str().to_owned(),
+                        message,
+                    })
+                }
+                kind => PathDeserializationError::new(kind),
```

Comment:
This is very nitty so take it or leave it, but if you really care just about one variant, you can use `if let` you'll have less indented code (maybe not in this case because the match is on the same line as the closure argument) and in the else branch you can directly return `e` instead of constructing the error again from a kind.